### ISSUE TITLE
Add support for gradient descent

### ISSUE THREAD

**Issue by jrgood01** [jrgood01/HaygoodAI#7]
Add support for gradient descent to train multi-layer neural networks.
